lib/ansible/modules/cloud/amazon/ec2_remote_facts.py

```diff
@@ -108,9 +108,11 @@ def get_instance_info(instance):
     except AttributeError:
         pass
 
+    instance_profile = dict(instance.instance_profile) if instance.instance_profile is not None else None
```

Comment:
What if `instance.instance_profile` is an empty dictionary?  I'm not sure I'd want to overwrite it w/ None.